internal/controllers/gateway/httproute_controller.go

```diff
@@ -585,3 +600,101 @@ func (r *HTTPRouteReconciler) getHTTPRouteRuleReason(ctx context.Context, httpRo
 	}
 	return gatewayv1beta1.RouteReasonResolvedRefs, nil
 }
+
+// ensureParentsAcceptedCondition sets the "Accepted" condition of HTTPRoute status.
+func (r *HTTPRouteReconciler) ensureParentsAcceptedCondition(
+	ctx context.Context,
+	httproute *gatewayv1beta1.HTTPRoute,
+	gateways []supportedGatewayWithCondition,
+	conditionStatus metav1.ConditionStatus,
+	conditionReason string,
+	conditionMessage string,
+) (bool, error) {
+	// map the existing parentStatues to avoid duplications
+	parentStatuses := make(map[string]*gatewayv1beta1.RouteParentStatus)
+	for _, existingParent := range httproute.Status.Parents {
+		namespace := httproute.Namespace
+		if existingParent.ParentRef.Namespace != nil {
+			namespace = string(*existingParent.ParentRef.Namespace)
+		}
+		existingParentCopy := existingParent
+		var sectionName string
+		if existingParent.ParentRef.SectionName != nil {
+			sectionName = string(*existingParent.ParentRef.SectionName)
+		}
+		parentStatuses[fmt.Sprintf("%s/%s/%s", namespace, existingParent.ParentRef.Name, sectionName)] = &existingParentCopy
+	}
```

Comment:
Since this is pretty much a copy paste of https://github.com/Kong/kubernetes-ingress-controller/blob/04f4d3b882a3f32402c9d3a7887ce2ba50c38e3f/internal/controllers/gateway/httproute_controller.go#L379-L392 how about we make this a function that accepts e.g. 
 `[]RouteParentStatus` and returns the parent statuses? We can then document it and perhaps add some unit tests for it. WDYT?

Reply:
I also found such code in processing other routes. So the function could be also adopted to other kind of routes. Because of the affect on all kinds of route, I think it is better to be done in another PR.

Reply:
Can we make a tracking issue for this so that we do not forget about this?

Reply:
OK, #3195